toolchain/check/decl_name_stack.h

```diff
@@ -237,10 +237,14 @@ class DeclNameStack {
   auto AddNameOrDiagnose(NameContext name_context, SemIR::InstId target_id,
                          SemIR::AccessKind access_kind) -> void;
 
-  // Adds a name to name lookup, or returns the existing instruction if this
+  // Adds a name to name lookup if not already declared or poisoned in this
+  // scope. If declared, returns the existing instruction and false. If
+  // poisoned, returns invalid instruction and true.
```

Comment:
"returns invalid instruction" feels like it's missing "an", how about one of:
```suggestion
  // poisoned, returns an invalid instruction and true.
```
```suggestion
  // poisoned, returns invalid and true.
```

Reply:
Added "an".
FWIW, I thought of "invalid instruction" as a term, but perhaps I should have used "InstId::Invalid" instead.